lib/util/trip/trip.dart

```diff
@@ -22,6 +22,10 @@ abstract class Trip extends Model {
   bool get isOngoing => startTime.isBefore(DateTime.now()) && endTime.isAfter(DateTime.now());
 
   bool overlapsWith(Trip other) {
-    return startTime.isBefore(other.endTime) && endTime.isAfter(other.startTime);
+    return overlapsWithTime(other.startTime, other.endTime);
+  }
+
+  bool overlapsWithTime(DateTime start, DateTime end) {
```

Comment:
this could just be a method overload of "overlapsWith". And it's probably nicer to just give it a time range instead? https://api.flutter.dev/flutter/material/DateTimeRange-class.html

Reply:
yes thats a good idea thanks

Reply:
i think dart don't allows overloading.

Reply:
ah okay good to know :D